ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java

```diff
@@ -649,16 +652,47 @@ public synchronized void measure(int tag, int[] outputBuffer) {
     if (rootView == null) {
       throw new NoSuchNativeViewException("Native view " + tag + " is no longer on screen");
     }
-    rootView.getLocationInWindow(outputBuffer);
+    computeBoundingBox(rootView, outputBuffer);
     int rootX = outputBuffer[0];
     int rootY = outputBuffer[1];
+    computeBoundingBox(v, outputBuffer);
+    outputBuffer[0] -= rootX;
+    outputBuffer[1] -= rootY;
+  }
 
-    v.getLocationInWindow(outputBuffer);
+  void computeBoundingBox(View v, int[] outputBuffer) {
+    mBoundingBox.set(0, 0, v.getWidth(), v.getHeight());
+    mapRectFromViewToWindowCoords(v, mBoundingBox);
 
-    outputBuffer[0] = outputBuffer[0] - rootX;
-    outputBuffer[1] = outputBuffer[1] - rootY;
-    outputBuffer[2] = v.getWidth();
-    outputBuffer[3] = v.getHeight();
+    outputBuffer[0] = (int) mBoundingBox.left;
+    outputBuffer[1] = (int) mBoundingBox.top;
+    outputBuffer[2] = (int) (mBoundingBox.right - mBoundingBox.left);
+    outputBuffer[3] = (int) (mBoundingBox.bottom - mBoundingBox.top);
+  }
+
+  // simplified version of the hidden Android method View.mapRectFromViewToScreenCoords()
+  void mapRectFromViewToWindowCoords(View v, RectF rect) {
+    Matrix m = v.getMatrix();
+    if (!m.isIdentity()) {
+      m.mapRect(rect);
+    }
+
+    rect.offset(v.getLeft(), v.getTop());
+
+    ViewParent parent = v.getParent();
+    while (parent instanceof View) {
+      View parentView = (View) parent;
+
+      rect.offset(-parentView.getScrollX(), -parentView.getScrollY());
+      m = parentView.getMatrix();
+      if (!m.isIdentity()) {
+        m.mapRect(rect);
+      }
+
+      rect.offset(parentView.getLeft(), parentView.getTop());
+
+      parent = parentView.getParent();
+    }
```

Comment:
I've commented the original Android code, explaining each change. 
Most of the changes are about replacing private access with a public method.
The most important comment is the last one.

```java
public void mapRectFromViewToScreenCoords(RectF rect, boolean clipToParent) {
    // hasIdentityMatrix is not public, instead I used getMatrix().isIdentity()
    if (!hasIdentityMatrix()) {
        getMatrix().mapRect(rect);
    }
    // mLeft & mTop are not public, instead I used getLeft() & getTop()
    rect.offset(mLeft, mTop);

    ViewParent parent = mParent;
    while (parent instanceof View) {
        View parentView = (View) parent;
        
        // mScrollX & mScrollY are not public, instead I used getScrollX() & getScrollY()
        rect.offset(-parentView.mScrollX, -parentView.mScrollY);
        
        // We don't need clipping to measure, so I removed this block entirely
        if (clipToParent) {
            rect.left = Math.max(rect.left, 0);
            rect.top = Math.max(rect.top, 0);
            rect.right = Math.min(rect.right, parentView.getWidth());
            rect.bottom = Math.min(rect.bottom, parentView.getHeight());
        }

        // hasIdentityMatrix is not public, instead I used getMatrix().isIdentity()
        if (!parentView.hasIdentityMatrix()) {
            parentView.getMatrix().mapRect(rect);
        }

        // mLeft & mTop are not public, instead I used getLeft() & getTop()
        rect.offset(parentView.mLeft, parentView.mTop);

        // mParent is not public, instead I used getParent()
        parent = parentView.mParent;
    }
    // I think the 2 following offsets would be the same for the measured view and 
    // the React root view. Since we subtract the root view position from the 
    // measured view position in measure(), these offsets are cancelled.
    // In short: (a + offsets) - (b + offsets) = a - b
    // 
    // class ViewRootImpl is hidden, I could not find a way to rewrite this code.
    // Since we don't need it for our purpose, I removed it.
    if (parent instanceof ViewRootImpl) {
        ViewRootImpl viewRootImpl = (ViewRootImpl) parent;
        rect.offset(0, -viewRootImpl.mCurScrollY);
    }
    rect.offset(mAttachInfo.mWindowLeft, mAttachInfo.mWindowTop);
}
```

Reply:
We definitely don't need `rect.offset(mAttachInfo.mWindowLeft, mAttachInfo.mWindowTop);` here since it actually offsets the coordinates relative to the screen coordinates, which isn't what we want (we want the coordinates relative to the window coordinates).